graphql/provider.go

```diff
@@ -36,14 +50,58 @@ func Provider() *schema.Provider {
 }
 
 func graphqlConfigure(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
+	diags := diag.Diagnostics{}
+
 	config := &graphqlProviderConfig{
 		GQLServerUrl:   d.Get("url").(string),
 		RequestHeaders: d.Get("headers").(map[string]interface{}),
 	}
-	return config, diag.Diagnostics{}
+
+	oauth2LoginQuery := d.Get("oauth2_login_query").(string)
+	oauth2LoginQueryVariables := d.Get("oauth2_login_query_variables").(map[string]interface{})
+	oauth2LoginQueryValueAttribute := d.Get("oauth2_login_query_value_attribute").(string)
+
+	if oauth2LoginQuery != "" && len(oauth2LoginQueryVariables) > 0 && oauth2LoginQueryValueAttribute != "" {
+		queryResponse, _, err := queryExecute(ctx, d, config, "oauth2_login_query", "oauth2_login_query_variables")
+		if err != nil {
+			return nil, diag.FromErr(fmt.Errorf("unable to execute oauth2_login_query: %w", err))
+		}
+
+		if queryErrors := queryResponse.ProcessErrors(); queryErrors.HasError() {
+			return nil, *queryErrors
+		}
+
+		var value string
+		if value, err = getOAuth2LoginQueryAttributeValue(oauth2LoginQueryValueAttribute, queryResponse.Data); err != nil {
+			return nil, diag.FromErr(err)
+		}
+
+		config.RequestAuthorizationHeaders = map[string]interface{}{
+			"Authorization": fmt.Sprintf("Bearer %s", value),
+		}
+	} else if oauth2LoginQuery != "" || len(oauth2LoginQueryVariables) > 0 || oauth2LoginQueryValueAttribute != "" {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "Incomplete OAuth 2.0 provider configuration",
+			Detail:   "All three attributes must be set: `oauth2_login_query`, `oauth2_login_query_variables` and `oauth2_login_query_value_attribute`.",
+		})
+	}
+
+	return config, diags
 }
 
 type graphqlProviderConfig struct {
 	GQLServerUrl   string
 	RequestHeaders map[string]interface{}
+
+	RequestAuthorizationHeaders map[string]interface{}
+}
+
+func getOAuth2LoginQueryAttributeValue(attribute string, data map[string]interface{}) (string, error) {
+	resourceKeyArgs := buildResourceKeyArgs(attribute)[1:] // Drop the leading `data` segment
```

Comment:
This is needed because `buildResourceKeyArgs()` adds the `data` segment but it isn't there in `queryResponse.Data`.

Reply:
Ahh.. this will do. Prior to having the queryResponse struct I was just using empty interface, but after I introduced the nicer struct I never updated the buildResourceKeyArgs func to use it. That is okay tho, it’s outside the scope of this PR and there are some other things I have to muck around with to make that work. Thanks 😊